topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.md

```diff
@@ -0,0 +1,521 @@
+Let's now add the replicate column which tells us which cells are part of pools of the same patient and tumor location.
+
+> <hands-on-title>Create replicate metadata</hands-on-title>
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3) %} *in a specific column* with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `Observation data`
+>    - *"1: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `2|4|7|8`
+>         - *"Replace with"*: `poolA`
+>    - **+ Insert Replacement**
+>    - *"2: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `3|5|9`
+>         - *"Replace with"*: `poolB`
+>    - **+ Insert Replacement**
+>    - *"3: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `6`
+>         - *"Replace with"*: `poolC`
+>    - **+ Insert Replacement**
+>    - *"4: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `0|1`
+>         - *"Replace with"*: `NA`
+>    - **+ Insert Replacement**
+>    - *"5: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `batch`
+>         - *"Replace with"*: `replicate`
+>
+> 2. {% tool [Cut](Cut1) %} with the following parameters:
+>    - *"Cut columns"*: `c2`
+>    - *"Delimited by"*: `Tab`
+>    - {% icon param-file %} *"From"*: `output of Replace Text`
+>
+> 3. **Rename** {% icon galaxy-pencil %} output `Replicate Metadata`
+>
+{: .hands_on}
+
+Next we will add the metadata indicating which patient each row came from.
+
+> <hands-on-title>Create patient data</hands-on-title>
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `Observation data`
+>    - *"1: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `0|1`
+>         - *"Replace with"*: `patient1`
+>    - **+ Insert Replacement**
+>    - *"2: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `2|3|4|5|6`
+>         - *"Replace with"*: `patient2`
+>    - **+ Insert Replacement**
+>    - *"3: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `7|8|9`
+>         - *"Replace with"*: `patient3`
+>    - **+ Insert Replacement**
+>    - *"5: Replacement"*
+>         - *"in column"*: `Column: 2`
+>         - *"Find pattern"*: `batch`
+>         - *"Replace with"*: `patient`
+>
+> 2. {% tool [Cut](Cut1) %} with the following parameters:
+>    - *"Cut columns"*: `c2`
+>    - *"Delimited by"*: `Tab`
+>    - {% icon param-file %} *"From"*: `output of Replace Text`
+>
+> 3. **Rename** {% icon galaxy-pencil %} output `Patient Metadata`
+>
+{: .hands_on}
+
+We will now add a column to indicate which sample each row came from using the sample ID's described earlier.
```

Comment:
Such a big command should be rewarded with some kind of image or screenshot or something to engage the user again.